worker/app/tasks/base.py

```diff
@@ -43,7 +43,7 @@ def set_up(self):
 
     def get_dns(self):
```

Comment:
I didn't go into too much detail, but I feel like `set_up` and `get_dns` has danger to be calling each other.

Reply:
right, thanks. even if it's safe and doesn't look like it is, it's wrong. refactored it, I think it's much better now.